KOGITO-8145 Fix broken links in SWF Guide - section Security (#234)
kiegroup/kogito-docs: serverlessworkflow/modules/ROOT/pages/security/authention-support-for-openapi-services.adoc

}
----

> If the OpenAPI specification file contains `securitySchemes` definitions, but not link:{open_api_spec_url}#security-requirement-object[Security Requirement Object] definitions, the generator can be configured to create the security requirement objects by default. In this case, for all operations without a security requirement, the default one is created. Note that the property value needs to match the name of a security scheme object definition, such as `http-basic-example` or `api-key-example` in the previous `securitySchemes` list.

Reviewer: ```suggestion
If the OpenAPI specification file contains `securitySchemes` definitions, but not the link:{open_api_spec_url}#security-requirement-object[Security Requirement Object] definitions, the generator is configured to create the security requirement objects by default. In this case, for all the operations without a security requirement, the default one is created. Note that the property value must match the name of a security scheme object definition, such as `http-basic-example` or `api-key-example` in the previous `securitySchemes` list.
```

}
----

> Unlike the `http basic`, `http bearer`, and `apiKey` security schemes, the OAuth 2.0 authentication relies on the link:https://quarkus.io/guides/security-openid-connect-client[Quarkus OpenId Connect (OIDC) and OAuth 2.0 Clients and Filters]. Therefore, you must add link:https://quarkus.io/guides/security-openid-connect-client-reference#oidc-client-filter[Quarkus OIDC Client Filter Extension] to your project as shown in the following example:

Reviewer: ```suggestion
Unlike the `http basic`, `http bearer`, and `apiKey` security schemes, the OAuth 2.0 authentication relies on the link:https://quarkus.io/guides/security-openid-connect-client[Quarkus OpenId Connect (OIDC) and OAuth 2.0 Clients and Filters]. Therefore, you must add the link:https://quarkus.io/guides/security-openid-connect-client-reference#oidc-client-filter[Quarkus OIDC Client Filter Extension] to your project as shown in the following example:
```
